### PR TITLE
Add sweep/spend-all to TxBuilder

### LINKED ISSUE #86: Update transaction building API to enable spend all / sweep
To send all / sweep all coins from a wallet add to the TxBuilder:

1. **TxBuilder drain_to(string address)** see `bdk` [wallet::tx_builder::TxBuilder::drain_to](https://docs.rs/bdk/latest/bdk/wallet/tx_builder/struct.TxBuilder.html#method.drain_to)

2. **TxBuilder drain_wallet()** see `bdk` [wallet::tx_builder::drain_wallet](https://docs.rs/bdk/latest/bdk/wallet/tx_builder/struct.TxBuilder.html#method.drain_wallet)

### PR DESCRIPTION
**Pull request by artfuldev**
Fixes #86 

* Adds `TxBuilder::drain_wallet`
* Adds `TxBuilder::drain_to`
* Introduces a `to_script_pubkey` helper
* Simplifies `TxBuilder::add_recipient` to use `to_script_pubkey` helper

### CODE REVIEW

File: src/bdk.udl [hunk 1]
```diff
@@ -144,6 +144,8 @@ interface TxBuilder {
   constructor();
   TxBuilder add_recipient(string address, u64 amount);
   TxBuilder fee_rate(float sat_per_vbyte);
+  TxBuilder drain_wallet();
+  TxBuilder drain_to(string address);
```
Comment (artfuldev) [inline src/bdk.udl:148]: Expose API

File: src/lib.rs [hunk 2]
```diff
@@ -1,7 +1,7 @@
 use bdk::bitcoin::hashes::hex::ToHex;
 use bdk::bitcoin::secp256k1::Secp256k1;
 use bdk::bitcoin::util::psbt::PartiallySignedTransaction;
-use bdk::bitcoin::{Address, Network};
+use bdk::bitcoin::{Address, Network, Script};
```
Comment (artfuldev) [inline src/lib.rs:4]: Add import for helper

File: src/lib.rs [hunk 3]
```diff
@@ -277,16 +277,26 @@ fn restore_extended_key(
     })
 }
 
+fn to_script_pubkey(address: &str) -> Result<Script, BdkError> {
+    Address::from_str(address)
+        .map(|x| x.script_pubkey())
+        .map_err(|e| BdkError::Generic(e.to_string()))
+}
+
```
Comment (artfuldev) [inline src/lib.rs:285]: Helper to convert `&str` to `Script` allowing failures

**Comment by artfuldev** [comment]
@notmandatory After this is merged and tagged/released I can pick up and close bitcoindevkit/bdk-kotlin#40

**Review by notmandatory — APPROVED** [review]
tested ACK 5512b31969b88cc352325683538d059d22dc67f3

I did a quick test with bdk-kotlin and it's working as expected.

**Comment by artfuldev** [comment]
@notmandatory have we tagged a release with this commit?

**Comment by notmandatory** [comment]
I was waiting for an approval on #126 but it's OK, I merged it now. 

Tag [v0.5.0](https://github.com/bitcoindevkit/bdk-ffi/releases/tag/v0.5.0) created.

**Pull request merged**